File: src/modules/grid-server/GridServer/Channels/Packets/Textfilter/TextFilterRequestPacket.cs
using Phoenix.Server;
using Phoenix.Server.Players;
using Phoenix.Common.Tasks;
using Phoenix.Common.Logging;
using System;
using Phoenix.Server.Configuration;
using Phoenix.Common;
using System.Collections.Generic;
using Phoenix.Common.Networking.Channels;
using Phoenix.Common.Networking.Packets;
using Phoenix.Common.IO;

namespace GridServer.Channels
{
    public class TextFilterRequestPacket : AbstractNetworkPacket
    {
        public string message;
        public bool strictChat;

        public override AbstractNetworkPacket Instantiate()
        {
            return new TextFilterRequestPacket();
        }

        public override void Parse(DataReader reader)
        {
            message = reader.ReadString();
            strictChat = reader.ReadBoolean();
        }

        public override void Write(DataWriter writer)
        {
            throw new NotImplementedException();
        }
    }
}

File: src/modules/grid-client/src/main/java/org/asf/edge/modules/gridclient/grid/channels/packets/textfilter/TextFilterRequestPacket.java
package org.asf.edge.modules.gridclient.grid.channels.packets.textfilter;

import java.io.IOException;

import org.asf.edge.modules.gridclient.phoenix.networking.DataReader;
import org.asf.edge.modules.gridclient.phoenix.networking.DataWriter;
import org.asf.edge.modules.gridclient.phoenix.networking.packets.IPhoenixPacket;

public class TextFilterRequestPacket implements IPhoenixPacket {

	public String message;
	public boolean strictChat;

	@Override
	public IPhoenixPacket instantiate() {
		return new TextFilterRequestPacket();
	}

	@Override
	public void parse(DataReader reader) throws IOException {
	}

	@Override
	public void build(DataWriter writer) throws IOException {
		writer.writeString(message);
		writer.writeBoolean(strictChat);
	}

}
